apps/opik-backend/src/main/java/com/comet/opik/domain/filter/FilterQueryBuilder.java

```diff
@@ -40,14 +40,16 @@ public class FilterQueryBuilder {
     private static final String USAGE_PROMPT_TOKENS_ANALYTICS_DB = "usage['prompt_tokens']";
     private static final String USAGE_TOTAL_TOKENS_ANALYTICS_DB = "usage['total_tokens']";
     private static final String VALUE_ANALYTICS_DB = "value";
+    private static final String CUSTOM_FIELD_ANALYTICS_DB = "data";
 
     private static final Map<Operator, Map<FieldType, String>> ANALYTICS_DB_OPERATOR_MAP = new EnumMap<>(Map.of(
             Operator.CONTAINS, new EnumMap<>(Map.of(
                     FieldType.STRING, "ilike(%1$s, CONCAT('%%', :filter%2$d ,'%%'))",
                     FieldType.LIST,
                     "arrayExists(element -> (ilike(element, CONCAT('%%', :filter%2$d ,'%%'))), %1$s) = 1",
                     FieldType.DICTIONARY,
-                    "ilike(JSON_VALUE(%1$s, :filterKey%2$d), CONCAT('%%', :filter%2$d ,'%%'))")),
+                    "ilike(JSON_VALUE(%1$s, :filterKey%2$d), CONCAT('%%', :filter%2$d ,'%%'))",
+                    FieldType.CUSTOM_FIELD, "ilike(%1$s[:filterKey%2$d], CONCAT('%%', :filter%2$d ,'%%'))")),
```

Comment:
Unless I'm missing something, I'm not convinced about the need for a CUSTOM_FIELD type. Data should be of DICTIONARY type, so you can search by the top level values and also by any deeper key.

Reply:
I also considered it, but this would require a new filter API. Keeping the compatibility for the deprecated fields and allowing a new field where the filter "name" is dynamic would be very hard. At least I couldn't think of a way to do this. 

Also, For now, the filters required were = contains and < >, no?